### PR TITLE
Fixing MacOS 13 assemble workflows

### PR DESCRIPTION
**Pull request by reta**
### Description
Fixing MacOS 13 asseble workflows

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/15743

### Check List
- [ ] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@dblock mind please taking a look? thank you!

File: .github/workflows/assemble.yml [hunk 1]
```diff
@@ -30,8 +30,11 @@ jobs:
       - name: Setup docker (missing on MacOS)
         id: setup_docker
         if: runner.os == 'macos'
+        continue-on-error: true
         run: |
-          exit 0;
+          brew install docker colima coreutils
+          gtimeout 15m colima start
```
Comment (reta) [inline .github/workflows/assemble.yml:36]: To mitigate possible future issues with hanging step(s), capping it to 15m

**Review by peternied — APPROVED** [review]
Thanks for fixing this up.  Those workflows on the mac can take nearly an hour, yikes that is a long time for a build.

File: .github/workflows/assemble.yml (hunk 1 above)
Comment (peternied) [reply .github/workflows/assemble.yml:36]: @reta Thanks for limiting the time just in case, looks reasonable to me.

**Pull request merged**